addons/misra.py

```diff
@@ -1833,7 +1833,10 @@ def reportErrorIfVariableIsNotConst(variable, stringLiteral):
                         continue
 
                     parametersUsed = getArguments(token)
-                    for i in range(len(parametersUsed)):
+
+                    checkArgCount = len(functionDeclaration.argument) - 1 if len(parametersUsed) > len(functionDeclaration.argument) else len(parametersUsed)
+                    checkArgCount = max(checkArgCount, 0)
+                    for i in range(checkArgCount):
                         usedParameter = parametersUsed[i]
```

Comment:
maybe instead of the complex code above the for loop we can add this here in the loop:

    if i >= functionDeclaration.argument:
        continue